src/main/java/org/gridsuite/modification/server/NetworkModificationService.java

```diff
@@ -39,8 +41,11 @@ private Network getNetwork(UUID networkUuid) {
         }
     }
 
-    void changeSwitchState(UUID networkUuid, String switchId, String open) {
+    public Set<String> changeSwitchState(UUID networkUuid, String switchId, String open) {
```

Comment:
In this case, I think the code is better by not using listeners, instead just return if the switch was modifed or not

Reply:
Modification done